components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.html

```diff
@@ -0,0 +1,110 @@
+<chef-modal [visible]="visible" (closeModal)="closeCreateModal()">
+  <div class="text-right">
+    <h2 slot="title">Create Environment</h2>
+  </div>
+  <div class="flex-container">
+    <div class="main pages-body">
+      <app-infra-tab-change class="tabs-row" (tabChange)="tabChange($event)">
+        <app-infra-tab tabTitle="Details" [active]="detailsTab" [disabled]="false">
+          <form [formGroup]="firstFormGroup">
+            <div class="input-margin">
+              <chef-form-field>
+                <label>
+                  <span class="label">Name</span>
+                  <input chefInput formControlName="name" type="text" id="name-input" (keyup)="handleInput($event)" autocomplete="off"/>
+                </label>
+                <chef-error
+                  *ngIf="(firstFormGroup.get('name').hasError('required') || firstFormGroup.get('name').hasError('pattern')) && firstFormGroup.get('name').dirty">
+                  Display Name is required.
+                </chef-error>
+                <chef-error *ngIf="conflictError">
+                  Environment "{{firstFormGroup.get('name').value}}" already exists.
+                </chef-error>
+              </chef-form-field>
+            </div>
+            <div class="input-margin">
+              <chef-form-field>
+                <label>
+                  <span class="label">Description</span>
+                  <textarea chefInput name="description" id="description" formControlName="description" autofocus></textarea>
+                </label>
+                <chef-error
+                  *ngIf="(firstFormGroup.get('description').hasError('required') || firstFormGroup.get('description').hasError('pattern')) && firstFormGroup.get('description').dirty">
+                  Description is required.
+                </chef-error>
+              </chef-form-field>
+            </div>
+            <div class="button-bar-box">
+              <chef-button tertiary (click)="closeCreateModal()">Cancel</chef-button>
+              <chef-button [disabled]="!firstFormGroup?.valid || conflictError"  primary id="create-button-object-modal" (click)="createEnvironment()">
+                <chef-loading-spinner *ngIf="creating"></chef-loading-spinner>
+                <span *ngIf="!creating">Create</span>
+                <span *ngIf="creating">Creating Environment...</span>
+              </chef-button>
+            </div>
+          </form>
+        </app-infra-tab>
+        <app-infra-tab tabTitle="Constraints" [active]="constraintsTab" [disabled]="!firstFormGroup?.valid">
+          <div *ngIf="showConstraint">
+            <app-infra-environment-constraint
+              [constraintKeys]="constraintKeysp"
+              (constraintValues)="constraintItemsHandler($event)"
+              [dynamicArray]="dynamicArrayp"
```

Comment:
Seems like you were trying to use "p" to differentiate names for both of these parameters here... no need to do that. Plus it makes it confusing to the reader.
```suggestion
              [constraintKeys]="constraintKeys"
              (constraintValues)="constraintItemsHandler($event)"
              [dynamicArray]="dynamicArray"
```
Actually, I would further suggest renaming "dynamicArray" on _both_ sides; that conveys no meaning to the reader. Should convey what is in the array or what it is used for.

Reply:
Agree with you, updated the changes.